meta-mel/conf/distro/include/mel.conf

```diff
@@ -178,11 +178,17 @@ require conf/distro/include/yocto-uninative.inc
 
 INHERIT += "uninative"
 ## Distro Features & Recipe Configuration {{{1
-MEL_DEFAULT_DISTRO_FEATURES = "largefile opengl multiarch vulkan"
+MEL_DEFAULT_DISTRO_FEATURES = "largefile opengl multiarch"
+
+# If EGLFS is enabled, vulkan is not required (depends on x11, wayland)
+MEL_DEFAULT_DISTRO_FEATURES += "${@'vulkan' if not bb.utils.to_boolean(d.getVar('ENABLE_EGLFS')) else ''}"
+
 # This violates typical MACHINE/DISTRO boundaries, but is part of MEL's
 # supported features. If the vendor supports x11 and not wayland for its
 # machines, so do we.
-MEL_DEFAULT_DISTRO_FEATURES += "${@bb.utils.contains('MACHINE_FEATURES', 'x11', bb.utils.contains('MACHINE_FEATURES', 'wayland', 'wayland', 'x11', d), 'wayland', d)}"
+MEL_DEFAULT_DISTRO_FEATURES += "${@'' if d.getVar('ENABLE_EGLFS', True) == '1' \
```

Comment:
Either use `== '1'` both here and on line 184, or use bb.utils.to_boolean both here and on line 184. Having two different semantics for the same variable within 10 lines isn't consistent.